File: oop-exercises/inheritance/nestjs/src/entity/CheckingAccount.ts
export class CheckingAccount {
  private balance: number;

  constructor(balance: number) {
    this.balance = balance;
  }

  public deposit(amount: number): void {
    this.balance += amount;
  }

  public withdraw(amount: number): void {
    if (this.balance < amount) {
      throw new Error("Insufficient balance to perform the transaction.");
    }
    this.balance -= amount;
  }

  public writeCheck(amount: number): void {
    if (this.balance < amount) {
      throw new Error("Insufficient balance to perform the transaction.");
    }
    this.balance -= amount;
  }

  public getBalance(): number {
    return this.balance;
  }
}

File: oop-exercises/inheritance/spring-boot/src/main/java/com/juandaabril/entity/CheckingAccount.java
package com.juandaabril.entity;

public class CheckingAccount {
    private double balance;

    public CheckingAccount(double balance) {
        this.balance = balance;
    }

    public void deposit(double amount) {
        balance += amount;
    }

    public void withdraw(double amount) {
        if (balance < amount) {
            throw new RuntimeException("Insufficient balance to perform the transaction.");
        }
        balance -= amount;
    }

    public void writeCheck(double amount) {
        if (balance < amount) {
            throw new RuntimeException("Insufficient balance to perform the transaction.");
        }

        balance -= amount;
    }

    public double getBalance() {
        return balance;
    }
}
